### PR TITLE
Support multiple Alicloud disks when creating machines

### PR DESCRIPTION
**Pull request by rfranzke**
**How to categorize this PR?**

/area storage
/kind enhancement
/priority normal
/platform alicloud

**What this PR does / why we need it**:
After https://github.com/gardener/gardener/pull/1893 was merged quite some time ago, this PR now adds support for the Alicloud extension to handle multiple Alicloud disks for the worker machines.

**Which issue(s) this PR fixes**:
Ref https://github.com/gardener/machine-controller-manager/issues/462

**Special notes for your reviewer**:
* Thanks @guydaichs for your work and contributions!

**Release note**:

```noteworthy user
It is now possible to configure additional data volumes for the worker machines. Please consult [this documentation](https://github.com/gardener/gardener-extension-provider-alicloud/blob/master/docs/usage-as-end-user.md#workerconfig) for more information.
```

### CODE REVIEW

File: example/30-worker.yaml [hunk 1]
```diff
@@ -96,5 +96,10 @@ spec:
     volume:
       type: cloud_efficiency
       size: 30Gi
+  # dataVolumes:
+  # - name: kubelet-dir
+  #   type: standard
```
Comment (ialidzhikov) [inline example/30-worker.yaml:101]:
```suggestion
  #   type: cloud_efficiency
```

File: pkg/apis/alicloud/validation/shoot.go [hunk 2]
```diff
@@ -51,6 +60,22 @@ func ValidateWorkers(workers []core.Worker, zones []apisalicloud.Zone, fldPath *
...
+			allErrs = append(allErrs, field.TooMany(fldPath.Index(i).Child("dataVolumes"), length, maxDataDiskCount))
+		}
+		for j, volume := range worker.DataVolumes {
+			dataVolPath := fldPath.Index(i).Child("dataVolumes").Index(j)
+			if volume.Name != nil && !dataDiskNameRegexp.MatchString(*volume.Name) {
+				allErrs = append(allErrs, field.Invalid(dataVolPath.Child("name"), *volume.Name, utilvalidation.RegexError(fmt.Sprintf("disk name given: %s does not match the expected pattern", *volume.Name), dataDiskNameFmt)))
+			} else if volume.Name != nil && len(*volume.Name) > 64 {
+				allErrs = append(allErrs, field.TooLong(dataVolPath.Child("name"), *volume.Name, 64))
+			}
+			allErrs = append(allErrs, validateVolume(volume.DeepCopy(), dataVolPath)...)
```
Comment (ialidzhikov) [inline pkg/apis/alicloud/validation/shoot.go:78]:
```suggestion
			allErrs = append(allErrs, validateVolume(volume, dataVolPath)...)
```

File: pkg/apis/alicloud/validation/shoot.go [hunk 3]
```diff
@@ -51,6 +60,22 @@ func ValidateWorkers(workers []core.Worker, zones []apisalicloud.Zone, fldPath *
...
+		if length := len(worker.DataVolumes); length > maxDataDiskCount {
+			allErrs = append(allErrs, field.TooMany(fldPath.Index(i).Child("dataVolumes"), length, maxDataDiskCount))
+		}
+		for j, volume := range worker.DataVolumes {
+			dataVolPath := fldPath.Index(i).Child("dataVolumes").Index(j)
+			if volume.Name != nil && !dataDiskNameRegexp.MatchString(*volume.Name) {
+				allErrs = append(allErrs, field.Invalid(dataVolPath.Child("name"), *volume.Name, utilvalidation.RegexError(fmt.Sprintf("disk name given: %s does not match the expected pattern", *volume.Name), dataDiskNameFmt)))
+			} else if volume.Name != nil && len(*volume.Name) > 64 {
+				allErrs = append(allErrs, field.TooLong(dataVolPath.Child("name"), *volume.Name, 64))
+			}
```
Comment (jia-jerry) [inline pkg/apis/alicloud/validation/shoot.go:77]: volume.Name can't be empty for data volumes

Comment (rfranzke) [reply pkg/apis/alicloud/validation/shoot.go:77]: What do you mean, Jerry? See also https://github.com/gardener/gardener-extension-provider-azure/pull/107#discussion_r438085401

Comment (jia-jerry) [reply pkg/apis/alicloud/validation/shoot.go:77]: > What do you mean, Jerry? See also [gardener/gardener-extension-provider-azure#107 (comment)](https://github.com/gardener/gardener-extension-provider-azure/pull/107#discussion_r438085401)

I just added a comment. If `volume.Name` keeps as a pointer, we need to make sure it is not a `null` value.

Comment (rfranzke) [reply pkg/apis/alicloud/validation/shoot.go:77]: As said, this is validated by the Gardener already. It cannot reach the system without being set.

**Pull request merged**